backend/lib/kubernetes-client/resources/Resource.js

```diff
@@ -0,0 +1,167 @@
+//
+// Copyright (c) 2019 by SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+'use strict'
+
+const { URLSearchParams } = require('url')
+const { join } = require('path')
+const { EventEmitter } = require('events')
+const pRetry = require('p-retry')
+const pEvent = require('p-event')
+const WebSocket = require('ws')
+const inject = require('reconnect-core')
+
+const BaseResource = require('./BaseResource')
+const { http, ws } = require('./symbols')
+
+const { setPatchType, wrapWebSocket, waitFor } = require('../util')
+
+class Resource extends BaseResource {
+  constructor (options = {}) {
+    super({ ...options, responseType: 'json' })
+  }
+
+  [http.prefixUrl] (url) {
+    let prefixUrl = super[http.prefixUrl](url)
+    const { group, version } = this.constructor
+    if (group === 'core' || group === '') {
+      prefixUrl += '/api/' + version
+    } else {
+      prefixUrl += '/apis/' + group + '/' + version
+    }
+    return prefixUrl
+  }
+
+  [http.pathname] ({ namespace, allNamespaces, name }) {
```

Comment:
split get into multiple functions with different names according to kubedoc https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.14/#-strong-read-operations-pod-v1-core-strong-

Reply:
✔️